### ISSUE TITLE
Debuging the transaction that calls method of non existing code exits with panic

### ISSUE THREAD

**Issue by radeksvarz** [foundry-rs/foundry#6710]
### Component

Cast, Anvil

### Have you ensured that all of these are up to date?

- [X] Foundry
- [ ] Foundryup

### What version of Foundry are you on?

*No response*

### What command(s) is the bug in?

cast send --debug

### Operating System

Linux

### Describe the bug

foundryup: installed - forge 0.2.0 (6fc7463 2024-01-05T00:21:51.712802723Z)
foundryup: installed - cast 0.2.0 (6fc7463 2024-01-05T00:21:51.718643672Z)
foundryup: installed - anvil 0.2.0 (6fc7463 2024-01-05T00:21:51.706678919Z)
foundryup: installed - chisel 0.2.0 (6fc7463 2024-01-05T00:21:51.726644879Z)

running towards local anvil

```
💤 cast send 0x70997970C51812dc3A010C7d01b50e0d17dc79C8 "increment()" --private-key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
 
blockHash 0xb5724628d1d2fc8ae236076f5a6d6a0d094418ec95b4a8f929fddcec9271acd8
blockNumber 1
contractAddress
cumulativeGasUsed 21064
effectiveGasPrice 4000000000
gasUsed 21064
logs []
logsBloom 0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
root
status 1
transactionHash 0x701aade4f053b0aa92c248182b9bcc1c4e5b22ad6caa1eb9704ad9e171eafdfe
transactionIndex 0
type 2
 
💤 cast run 0x701aade4f053b0aa92c248182b9bcc1c4e5b22ad6caa1eb9704ad9e171eafdfe
Executing previous transactions from the block.
Traces:
  [0] 0x70997970C51812dc3A010C7d01b50e0d17dc79C8::increment()
    └─ ← ()
 
 
Transaction successfully executed.
Gas used: 21064
 
💤 cast run 0x701aade4f053b0aa92c248182b9bcc1c4e5b22ad6caa1eb9704ad9e171eafdfe --debug
Executing previous transactions from the block.
The application panicked (crashed).
Message: index out of bounds: the len is 0 but the index is 0
Location: crates/debugger/src/tui/context.rs:64
 
This is a bug. Consider reporting it at https://github.com/foundry-rs/foundry
 
Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
Aborted (core dumped)
 
```

**Comment by DaniPopes** [comment]
Panic was fixed, and replaced by an error, although I don't really know why these errors happen:

```
...
$ cast run 0x701aade4f053b0aa92c248182b9bcc1c4e5b22ad6caa1eb9704ad9e171eafdfe --debug
Executing previous transactions from the block.
Error:
debug arena is empty
```

**Comment by cakevm** [comment]
I ran in a similar issue where it was reporting `debug arena is empty`. The cause for me was that the security setting from MacOS prevented the network access for my shell. I only found out when using anvil `anvil --fork-url <url>` reported `No route to host`. I am using iTerm2 and [resetting the security settings](https://apple.stackexchange.com/questions/476030/no-route-to-host-on-iterm2-only) resolve my issue. 

Solution:
Run `sudo tccutil reset All com.googlecode.iterm2`, open iTerm2 again and accept all prompts.
